tests/queryset_pickle/tests.py

```diff
@@ -153,3 +153,21 @@ def test_unsupported_unpickle(self):
         msg = "Pickled queryset instance's Django version 1.0 does not match the current version %s." % get_version()
         with self.assertRaisesMessage(RuntimeWarning, msg):
             pickle.loads(pickle.dumps(qs))
+
+    def test_queryset_evaluation(self):
+        """
+        #27159 -- Test that no unintentional queryset evaluation happens
```

Comment:
Just state the expected behavior such as "Pickling a QuerySet with an `__in=inner_qs` lookup, shouldn't evaluate inner_qs." We're reserving ticket references for obscure issues that can't be easily described in docstrings (the above tests aren't a good example, unfortunately).